library/core/src/slice/mod.rs

```diff
@@ -3876,6 +3876,18 @@ impl<T> [T] {
         } else {
             let (left, rest) = self.split_at(offset);
             let (us_len, ts_len) = rest.align_to_offsets::<U>();
+            // Inform Miri that we want to consider the "middle" pointer to be suitably aligned.
+            #[cfg(miri)]
+            {
+                extern "Rust" {
+                    pub fn miri_promise_symbolic_alignment(ptr: *const (), align: usize);
```

Comment:
Can this be declared only once? For instance in `core::intrinsics`?

Reply:
Since it is so Miri-specific I wanted to avoid that, but sure, can do.